docs/software-architecture-and-design.md

```diff
@@ -457,7 +462,16 @@ Although we want to display information about the [AI](#ai) in the [Visualizer](
 A [DrawFunction](#draw_functions) is essentially a function that tells the [Visualizer](#visualizer) _how_ to draw something. When created, [DrawFunctions](#draw_functions) use [lazy-evaluation](https://www.tutorialspoint.com/functional_programming/functional_programming_lazy_evaluation.htm) to embed the data needed for drawing into the function itself. What is ultimately produced is a function that the [Visualizer](#visualizer) can call, with the data to draw (and the details of how to draw it) already included. This function can then be sent over the Observer system to the [Visualizer](#visualizer). The [Visualizer](#visualizer) can then run this function to perform the actual draw operation.
 
 
-# Simulated Integration Tests
+# Simulator
+The `Simulator` is what we use for physics simulation to do testing when we don't have access to real field. In terms of the architecture, the `Simulator` performs the functions of (1) [SSL-Vision](#ssl-vision) by publishing new vision data and (2) the robots by accepting new [Primitives](#primitives). Using the current state of the simulated world, the `Simulator` simulates the new [Primitives](#primitives) over some time step and publishes new ssl vision data based on the updated simulated world. The `Simulator` is designed to be "perfect", which means that (1) the vision data it publishes exactly reflects the state of the simulated world and (2) the simulation perfectly reflects our best understanding of the physics (e.g. friction) with no randomness.
```

Comment:
too lazy to check how this looks in rendered markdown, but the functions of the `Simulator` would be good as a numbered list (not all inline like this).

Reply:
made the points into lists